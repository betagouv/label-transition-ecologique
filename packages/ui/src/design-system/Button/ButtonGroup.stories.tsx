import {useState} from 'react';
import {Meta, StoryObj} from '@storybook/react';

import {ButtonGroup} from './ButtonGroup';

const meta: Meta<typeof ButtonGroup> = {
  component: ButtonGroup,
  args: {
    activeButtonId: '2',
    buttons: [
      {id: '1', children: 'Bouton 1'},
      {id: '2', children: 'Bouton 2'},
      {id: '3', children: 'Bouton 3'},
    ],
  },
  render: args => {
    const [activeId, setActiveId] = useState(args.activeButtonId);
    return (
      <ButtonGroup
        {...args}
        activeButtonId={activeId}
        buttons={args.buttons.map(button => ({
          ...button,
          onClick: () => setActiveId(button.id),
        }))}
      />
    );
  },
};

export default meta;

type Story = StoryObj<typeof ButtonGroup>;

export const Default: Story = {
  args: {},
};

export const Size: Story = {
  args: {
    size: 'xs',
  },
};

export const Disabled: Story = {
  args: {
    buttons: [
      {id: '1', children: 'Bouton 1', disabled: true},
      {id: '2', children: 'Bouton 2'},
      {id: '3', children: 'Bouton 3'},
    ],
  },
};

export const Deux: Story = {
  args: {
    buttons: [
      {id: '1', children: 'Bouton 1'},
      {id: '2', children: 'Bouton 2'},
    ],
  },
};
