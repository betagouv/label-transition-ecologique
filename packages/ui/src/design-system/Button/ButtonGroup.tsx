import {CSSProperties} from 'react';
import classNames from 'classnames';

import {DefaultButtonProps, ButtonSize} from './types';
import {Button} from './Button';

type Props = {
  /** Listes de boutons à afficher avec leurs propriétés */
  buttons: DefaultButtonProps[];
  /** Permet d'ajuster les styles du container */
  className?: string;
  /** Taille des boutons */
  size?: ButtonSize;
  /** Id du bouton affiché dans une variante primary */
  activeButtonId?: string;
};

/**
 *  Affiche un groupe d'onglets.
 */
export const ButtonGroup = ({
  className,
  size = 'md',
  buttons,
  activeButtonId,
}: Props) => {
  /** Applique les bons styles pour les borders et border-radius
   * en fonction de la position du bouton dans le tableau */
  const getButtonStyles = (index: number): CSSProperties => {
    if (index === 0) {
      return {
        borderTopRightRadius: 0,
        borderBottomRightRadius: 0,
      };
    } else if (buttons.length === 2) {
      return {
        borderTopLeftRadius: 0,
        borderBottomLeftRadius: 0,
        borderLeft: 0,
      };
    } else {
      if (index > 0 && index < buttons.length - 1)
        return {
          borderRadius: 0,
          borderRight: 0,
          borderLeft: 0,
        };
      if (index === buttons.length - 1)
        return {
          borderTopLeftRadius: 0,
          borderBottomLeftRadius: 0,
        };
    }
  };

  return (
    <div className={classNames('flex items-center', className)}>
      {buttons.map((props, index) => (
        <Button
          key={index}
          {...props}
          size={size}
          variant={activeButtonId === props.id ? 'primary' : 'outlined'}
          className="!border-primary-7"
          style={getButtonStyles(index)}
        />
      ))}
    </div>
  );
};
