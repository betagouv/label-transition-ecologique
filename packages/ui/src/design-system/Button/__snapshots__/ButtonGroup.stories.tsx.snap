// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design-system/Button/ButtonGroup Default smoke-test 1`] = `
<div class="flex items-center">
  <button id="1"
          class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6 !border-primary-7"
          style="border-top-right-radius: 0px; border-bottom-right-radius: 0px;"
  >
    Bouton 1
  </button>
  <button id="2"
          class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8 text-base py-3 px-6 !border-primary-7"
          style="border-radius: 0px; border-right: 0px; border-left: 0px;"
  >
    Bouton 2
  </button>
  <button id="3"
          class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6 !border-primary-7"
          style="border-top-left-radius: 0px; border-bottom-left-radius: 0px;"
  >
    Bouton 3
  </button>
</div>
`;

exports[`design-system/Button/ButtonGroup Deux smoke-test 1`] = `
<div class="flex items-center">
  <button id="1"
          class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6 !border-primary-7"
          style="border-top-right-radius: 0px; border-bottom-right-radius: 0px;"
  >
    Bouton 1
  </button>
  <button id="2"
          class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8 text-base py-3 px-6 !border-primary-7"
          style="border-top-left-radius: 0px; border-bottom-left-radius: 0px; border-left: 0px;"
  >
    Bouton 2
  </button>
</div>
`;

exports[`design-system/Button/ButtonGroup Disabled smoke-test 1`] = `
<div class="flex items-center">
  <button id="1"
          disabled
          class="w-fit flex items-center border-solid group gap-2 cursor-not-allowed rounded-lg border font-bold !text-primary-3 !bg-primary-0 !border-primary-4 text-base py-3 px-6 !border-primary-7"
          style="border-top-right-radius: 0px; border-bottom-right-radius: 0px;"
  >
    Bouton 1
  </button>
  <button id="2"
          class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8 text-base py-3 px-6 !border-primary-7"
          style="border-radius: 0px; border-right: 0px; border-left: 0px;"
  >
    Bouton 2
  </button>
  <button id="3"
          class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6 !border-primary-7"
          style="border-top-left-radius: 0px; border-bottom-left-radius: 0px;"
  >
    Bouton 3
  </button>
</div>
`;

exports[`design-system/Button/ButtonGroup Size smoke-test 1`] = `
<div class="flex items-center">
  <button id="1"
          class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !border-primary-7"
          style="border-top-right-radius: 0px; border-bottom-right-radius: 0px;"
  >
    Bouton 1
  </button>
  <button id="2"
          class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !border-primary-7"
          style="border-radius: 0px; border-right: 0px; border-left: 0px;"
  >
    Bouton 2
  </button>
  <button id="3"
          class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !border-primary-7"
          style="border-top-left-radius: 0px; border-bottom-left-radius: 0px;"
  >
    Bouton 3
  </button>
</div>
`;
